docs_devel/docs/12.HowToMakeMTConnector.md

```diff
@@ -0,0 +1,112 @@
+
+The skeleton project uses Gradle for a build system.
+You can select Groovy DSL or Kotlin DSL for the configuration.
+When you choose a Groovy DSL, please rename `build.gradle.disabled` to `build.gradle` and remove `build.gradle.kts`.
+
+You should also modify a file
+- Project.name in settings.gradle
+- Properties: description, title, website and category
+- Plugin Main class name in build.gradle(.kts).
+
+Implementation should be placed at
+- Source code: src/main/<lang>/*
+- Test code: src/test/<lang>/* and src/test/resources/*
+
+## OmegaT MT API - `org.omegat.gui.exttrans.IMachineTranslation`
+
+### String getTranslation(Language sLang, Language tLang, String text) throws Exception;
+
+It is a main method to override in your plugin project. The method recieves a source text, translation from sLang to 
+tLang language, and return translation text.
+
+### String getCachedTranslation(Language sLang, Language tLang, String text);
+
+It is another method that you should override. If your plugin has a caching mechanism, and there is entry for the text,
+return translation without accessing MT engine. Otherwise, return null.
+
+> **_NOTE:_**  You should not call the engine in the method, because OmegaT will call it even when the target 
+> segment has already translated, but configured as `MT only untranslated segment` is on or `automatically translate` is off.
+
+### getName()
+
+Human and machine-readable name of MT engine. OmegaT uses it for the index of results, and as a property value, and 
+logging purpose.
+
+### isEnabled()/setEnabled(boolean b)
+
+return and set an enabled status.
+
+### boolean isConfigurable() /  void showConfigurationUI(Window parent)
+
+When the plugin return true for the method, `isConfigurable()` OmegaT will call `showConfigurationUI` when user 
+clicks a configuration UI button on `Tools > Preferences > MachineTranslation`.
+
+## Convenience abstract class
+
+There are two convenience class for plugins. You are recommended to use `org.omegat.core.
+machinetranslators.BaseCachedTranslate`. It implements many necessary things, and you can concentrate into a logic to 
+access your MT engine API.  There are only three methods you should override.
+
+### getName();
+
+Name of the engine as same as `IMachineTranslate`.
+
+### protected abstract String getPreferenceName();
+
+A preference key used in a user's configuration file. It will be like a "allow_<ENGINE NAME>_translate".
+
+### protected abstract String translate(Language sLang, Language tLang, String text) throws Exception;
+
+Main method to return translation from source text.
+
+
+## Common way to register your plugin
+
+You can register your plugin through a Core method like as follows;
+
+```java
+    public static void loadPlugins() {
+        Core.registerMachineTranslationClass(PluginClassName.class);
+    }
+
+    public static void unloadPlugins() {
+    }
```

Comment:
We should add a constructor example how to initialize things in the connector. Also we should guide developer to avoid static initialization of the connector.